app/routes.py

```diff
@@ -1,2 +1,292 @@
-from flask import Blueprint
+from flask.wrappers import Response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request
+from datetime import datetime, timezone
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
 
+#create the blueprint for the endpoints
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#Create a Task: Valid Task With null completed_at
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    
+    new_task = set_task(request_body)
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {}
+    response['task'] = format_task_dictionary(new_task)
+    return make_response(response, 201)
+
+###################
+#Helper functions for tasks
+def set_task(request_body):
+    new_task = Task(title=request_body["title"],
+                description=request_body["description"])
+    
+    if request_body["completed_at"] is not None:
+        new_task.completed_at=request_body["completed_at"]
+    else:
+        new_task.completed_at = None
+    
+    return new_task
+
+def format_task_dictionary(new_task):
+    task = {}
+    task['id'] = new_task.task_id
+    task['title'] = new_task.title #request_body["title"]
+    task['description'] = new_task.description #request_body["description"]
+    if new_task.completed_at is not None:
+        task['is_complete'] = True
+    else:
+        task['is_complete'] = False
+
+    if new_task.goal_id is not None:
+        task['goal_id'] = new_task.goal_id
+
+    return task
+####################
+
+#Get Tasks: Getting Saved Tasks
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+
+    #this can go in a helper function
+    sort_query = request.args.get("sort")
+    if(sort_query == "asc"):
+        tasks = Task.query.order_by(Task.title).all()
+    elif(sort_query == "desc"):
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+    
+    response = []
+    for task in tasks:
+        task_dict = format_task_dictionary(task)
+        response.append(task_dict)
+
+    return jsonify(response)
+
+#Get a Task
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_task(task_id):
+    task = Task.query.get(task_id)
+    
+    if task is None:
+        return make_response("Not Found", 404)
```

Comment:
Just for consistency

```suggestion
        return make_response(f"Task {task_id} not found", 404)
```